docs/integrations/azure/org.md

```diff
@@ -258,67 +258,204 @@ The `myUserTransformer`, `myGroupTransformer`, `myOrganizationTransformer`, and
       children: [],
     },
   };
+  // highlight-add-end
 }
+```
 
-// This user transformer makes use of the built in logic, but also sets the description field
+This User Transformer example makes use of the built-in logic, but also modifies the username and sets a description
+
+```ts
 export async function myUserTransformer(
   graphUser: MicrosoftGraph.User,
   userPhoto?: string,
 ): Promise<UserEntity | undefined> {
   const backstageUser = await defaultUserTransformer(graphUser, userPhoto);
-
+  // highlight-add-start
+  // Make sure the default transformer returned an entity
   if (backstageUser) {
-    backstageUser.metadata.description = 'Loaded from Microsoft Entra ID';
-  }
+    // Update the description to make it obvious where this entity came from
+    backstageUser.metadata.description =
+      'Loaded from Microsoft Entra ID via MyCustomUserTransformer';
 
+    // The default transformer sets the username to the email address with invalid characters subbed out: 'user_domain.com'
+    // Set the username to the local part of the email address in lowercase without the domain
+    const newName = backstageUser.metadata.name.split('_')[0].toLowerCase();
+    backstageUser.metadata.name = newName;
+
+    return backstageUser;
+  }
+  return undefined;
+  // highlight-add-end
+  // highlight-remove-start
   return backstageUser;
+  // highlight-remove-end
 }
+```
+
+This Organization Transformer example removes the organization group completely by returning undefined
 
-// Example organization transformer that removes the organization group completely
+```ts
 export async function myOrganizationTransformer(
   graphOrganization: MicrosoftGraph.Organization,
 ): Promise<GroupEntity | undefined> {
+  // highlight-remove-start
+  const backstageOrg = await defaultOrganizationTransformer(graphOrganization);
+  return backstageOrg;
+  // highlight-remove-end
+  // highlight-add-start
+  // The org transformer creates a group to be used as the base of the relationship tree for groups
+  // We don't need this to be created, so return undefined instead of an entity
   return undefined;
+  // highlight-add-end
 }
+```
 
-// Example config transformer that expands the group filter to also include 'azure-group-a'
+This Config Transformer example expands the group filter to also include 'azure-group-a'
+
+```ts
 export async function myProviderConfigTransformer(
   provider: MicrosoftGraphProviderConfig,
 ): Promise<MicrosoftGraphProviderConfig> {
+  // highlight-add-start
+  // The filter in our config file relies on a property that has been intermittantly causing this important group to fail ingestion
+  // Ensure the group is always discovered by the filter
   if (!provider.groupFilter?.includes('azure-group-a')) {
     provider.groupFilter = `${provider.groupFilter} or displayName eq 'azure-group-a'`;
   }
+  // highlight-add-end
   return provider;
 }
 ```
 
+Now we just need to add our new module to the Backend.
+
+```ts
```

Comment:
```suggestion
```ts title="packages/backend/src/index.ts"
```

If you do that then you can remove the line below as it will be the title of the code example 👍